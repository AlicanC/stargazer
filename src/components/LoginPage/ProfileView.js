// @flow

import * as React from 'react';
import gql from 'graphql-tag';

type Props = {
  user: Object,
};

export default function ProfileView(props: Props) {
  const { user } = props;

  return (
    <div className="mb-3" style={{ textAlign: 'center' }}>
      <img
        className="mb-3"
        src={user.avatarUrl}
        alt={`${user.name}'s Avatar`}
        style={{ width: '60px', borderRadius: '30px' }}
      />
      <br />
      {user.name}
    </div>
  );
}

ProfileView.fragments = {
  user: gql`
    fragment ProfileViewUser on User {
      name
      avatarUrl
    }
  `,
};
