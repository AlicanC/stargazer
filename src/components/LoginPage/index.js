// @flow

import * as React from 'react';
import { compose, graphql } from 'react-apollo';
import gql from 'graphql-tag';
import { Container, Col, Row, Button } from 'reactstrap';
import { connect } from 'react-redux';

import ProfileView from './ProfileView';

type Props = {
  data: Object,
  dispatch: Object,
};

class LoginPage extends React.PureComponent<Props, void> {
  onLoginClick = () => {
    const { dispatch } = this.props;

    const token = prompt('Your GitHub token');

    if (!token) return;

    dispatch.login(token);
  };

  onLogoutClick = () => {
    const { dispatch } = this.props;

    dispatch.logout();
  };

  render() {
    const { data } = this.props;

    const user = data.viewer;

    return (
      <Container>
        {user ? (
          <>
            <Row className="mb-3">
              <Col className="text-center">
                <ProfileView user={user} />
              </Col>
            </Row>
            <Row>
              <Col className="text-center">
                <Button onClick={this.onLogoutClick} color="primary">
                  Logout
                </Button>
              </Col>
            </Row>
          </>
        ) : (
          <>
            <Row className="mb-3">
              <Col className="text-center">
                <h5>Login now to start hunting for superstar repos!</h5>
              </Col>
            </Row>
            <Row>
              <Col className="text-center">
                <p>
                  Create a GitHub token <a href="https://github.com/settings/tokens/new">here</a>{' '}
                  and login.
                </p>
                <Button onClick={this.onLoginClick} color="primary">
                  Login
                </Button>
              </Col>
            </Row>
          </>
        )}
      </Container>
    );
  }
}

export default compose(
  connect(),
  graphql(gql`
    query LoginPageQuery {
      viewer {
        id
        ...ProfileViewUser
      }
    }
    ${ProfileView.fragments.user}
  `),
)(LoginPage);
