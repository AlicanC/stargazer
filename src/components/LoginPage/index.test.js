import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';

import LoginPage from '.';

jest.mock('react-router-dom', () => ({
  Link: 'div',
}));

jest.mock('react-redux', () => ({
  connect: () => (C) => C,
}));

it('matches snapshot', () => {
  const renderer = TestRenderer.create(
    <MockedProvider>
      <LoginPage />
    </MockedProvider>,
  );

  expect(renderer.toJSON()).toMatchSnapshot();
});
