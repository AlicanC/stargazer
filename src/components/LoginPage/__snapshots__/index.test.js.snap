// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="mb-3 row"
  >
    <div
      className="text-center col"
    >
      <h5>
        Login now to start hunting for superstar repos!
      </h5>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="text-center col"
    >
      <p>
        Create a GitHub token 
        <a
          href="https://github.com/settings/tokens/new"
        >
          here
        </a>
         
        and login.
      </p>
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
